pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1069,6 +1071,64 @@ def test_dt64arr_add_sub_parr(
         )
         assert_invalid_addsub_type(dtarr, parr, msg)
 
+    def test_dt64arr_addsub_time_object_raises(self, box_with_array, tz_naive_fixture):
+        # https://github.com/pandas-dev/pandas/issues/10329
+
+        tz = tz_naive_fixture
+
+        obj1 = pd.date_range("2012-01-01", periods=3, tz=tz)
+        obj2 = [time(i, i, i) for i in range(3)]
+
+        obj1 = tm.box_expected(obj1, box_with_array)
+        obj2 = tm.box_expected(obj2, box_with_array)
+
+        with warnings.catch_warnings(record=True):
```

Comment:
can this be tm.assert_produces_warning?

Reply:
I really prefer not too, because I am not trying to validate that this produces warnings, the warnings here are just a side effect.